File: RandomPasswordGenerator.py
import random
import string

def generate_random_password(length, include_uppercase, include_lowercase, include_digits, include_special_chars):
    valid_chars = ""
    
    if include_uppercase:
        valid_chars += string.ascii_uppercase
    if include_lowercase:
        valid_chars += string.ascii_lowercase
    if include_digits:
        valid_chars += string.digits
    if include_special_chars:
        valid_chars += string.punctuation
    
    if not valid_chars:
        print("Erro: Nenhum tipo de caractere selecionado.")
        return
    
    password = ''.join(random.choice(valid_chars) for _ in range(length))
    return password

def main():
    print("Bem-vindo ao Gerador de Senhas Aleatórias!")
    
    while True:
        length = int(input("\nComprimento da senha: "))
        include_uppercase = input("Incluir letras maiúsculas? (S/N): ").strip().lower() == "s"
        include_lowercase = input("Incluir letras minúsculas? (S/N): ").strip().lower() == "s"
        include_digits = input("Incluir dígitos? (S/N): ").strip().lower() == "s"
        include_special_chars = input("Incluir caracteres especiais? (S/N): ").strip().lower() == "s"
        
        generated_password = generate_random_password(length, include_uppercase, include_lowercase, include_digits, include_special_chars)
        print(f"\nSenha gerada: {generated_password}\n")
        
        another_password = input("Deseja gerar outra senha? (S/N): ").strip().lower()
        if another_password != "s":
            break
    
    print("Obrigado por usar o Gerador de Senhas Aleatórias!")

if __name__ == "__main__":
    main()

File: RandomPasswordGenerator.cs
using System;
using System.Text;

namespace GeradorSenhasAleatorias
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Bem-vindo ao Gerador de Senhas Aleatórias!");
            
            while (true)
            {
                Console.Write("\nComprimento da senha: ");
                int length = int.Parse(Console.ReadLine());
                
                Console.Write("Incluir letras maiúsculas? (S/N): ");
                bool includeUppercase = Console.ReadLine().Trim().ToLower() == "s";
                
                Console.Write("Incluir letras minúsculas? (S/N): ");
                bool includeLowercase = Console.ReadLine().Trim().ToLower() == "s";
                
                Console.Write("Incluir dígitos? (S/N): ");
                bool includeDigits = Console.ReadLine().Trim().ToLower() == "s";
                
                Console.Write("Incluir caracteres especiais? (S/N): ");
                bool includeSpecialChars = Console.ReadLine().Trim().ToLower() == "s";
                
                string generatedPassword = GenerateRandomPassword(length, includeUppercase, includeLowercase, includeDigits, includeSpecialChars);
                Console.WriteLine($"\nSenha gerada: {generatedPassword}\n");
                
                Console.Write("Deseja gerar outra senha? (S/N): ");
                if (Console.ReadLine().Trim().ToLower() != "s")
                    break;
            }
            
            Console.WriteLine("Obrigado por usar o Gerador de Senhas Aleatórias!");
        }
        
        static string GenerateRandomPassword(int length, bool includeUppercase, bool includeLowercase, bool includeDigits, bool includeSpecialChars)
        {
            const string uppercaseChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
            const string lowercaseChars = "abcdefghijklmnopqrstuvwxyz";
            const string digitChars = "0123456789";
            const string specialChars = "!@#$%^&*()_-+=<>?";

            StringBuilder validChars = new StringBuilder();
            if (includeUppercase)
                validChars.Append(uppercaseChars);
            if (includeLowercase)
                validChars.Append(lowercaseChars);
            if (includeDigits)
                validChars.Append(digitChars);
            if (includeSpecialChars)
                validChars.Append(specialChars);

            Random random = new Random();
            StringBuilder password = new StringBuilder();

            for (int i = 0; i < length; i++)
            {
                int randomIndex = random.Next(0, validChars.Length);
                password.Append(validChars[randomIndex]);
            }

            return password.ToString();
        }
    }
}
